zap/src/output/luau/client.rs

```diff
@@ -199,62 +201,73 @@ impl<'src> ClientOutput<'src> {
 			self.push_stmts(&des::gen(data, "value", true));
 		}
 
-		if ev.call == EvCall::SingleSync || ev.call == EvCall::SingleAsync {
-			self.push_line(&format!("if events[{id}] then"));
-		} else {
-			self.push_line(&format!("if events[{id}][1] then"));
-		}
+		match ev.call {
+			EvCall::Polling => {
+				// Event types without data use `true` as a placeholder.
+				self.push_line(&format!(
+					"table.insert(polling_payload_queues[{id}], if value == nil then true else value)"
+				));
```

Comment:
if true is being used as a placeholder, then what happens to real true values? I recommend we use some "null" userdata to represent nil

Reply:
Or maybe we use a counter for events w/o data instead of an array - sorta like how it works already for queuing

Reply:
> Or maybe we use a counter for events w/o data instead of an array - sorta like how it works already for queuing

Something needs to be returned to the loop or iteration will end, there's no player to return on the client.

Reply:
> if true is being used as a placeholder, then what happens to real true values? I recommend we use some "null" userdata to represent nil

I don't see much benefit to this unless you assign a metatable that errors on everything, but you don't like metatables.

```luau
for meaningless_value in event do
    if meaningless_value then
        -- Same behavior either way.
    end
end
```

Reply:
You'd also need to lie to the type system because `newproxy(true)` returns `any`, which would be worse than `true`.

Reply:
we need to do something because if we use `true` then `boolean?` becomes impossible